### DISCUSSION TITLE
An issue with displaying available endpoints in DRF's default API root view

### DISCUSSION THREAD

**Discussion by varnie** [encode/django-rest-framework#8879] [Question & Answer]
Hello everybody.

I have the following `urls.py` for an app named service where I register API endpoints:
```
from .views import AccessViewSet, CheckViewSet

app_name = "api"

router = DefaultRouter()

router.register(r"access/(?P<endpoint>.+)", AccessViewSet, basename="access")
router.register(r"check/(?P<endpoint>.+)", CheckViewSet, basename="check")


urlpatterns = [
    path("", include(router.urls)),
]
```

Below is my project's `urls.py` where I use it:
```
from django.urls import include, path

import service.urls as service_urls

urlpatterns = [
    # ...
    path("service/", include('service.urls')),
]
```

The APIs themselves are functioning properly, but I am having trouble making them work with DRF's default API root view. The view is displaying an empty list of available endpoints. 
I'm not sure, but this issue may be related to the regular expressions I'm using when registering endpoints, such as `r"access/(?P<endpoint>.+)`.

The problem disappears if I rewrite it like `router.register(r"access", AccessViewSet, basename="access")` (i.e. I delete that part: `(?P<endpoint>.+)`).

If this is indeed the problem, how can I resolve it?

I am using:
> Django==4.1.5
> djangorestframework==3.14.0

Thank you for your time.

**Comment by auvipy** [comment]
can you please check the docs to ensure (?P<endpoint>.+) is needed after router.register(r"access/ ?

**Reply by varnie** [reply]
> can you please check the docs to ensure (?P.+) is needed after router.register(r"access/ ?

what do you mean? It is needed for the purpose of my APIs.

**Reply by auvipy** [reply]
because you are doing it wrong. check the docs https://www.django-rest-framework.org/api-guide/routers/ thanks

**Reply by varnie** [reply]
> because you are doing it wrong. check the docs https://www.django-rest-framework.org/api-guide/routers/ thanks

what am I doing wrong? thanks
